drivers/i2c/i2c_ll_stm32.c

```diff
@@ -526,9 +508,12 @@ static const struct i2c_stm32_config i2c_stm32_cfg_##index = {		\
 	STM32_I2C_IRQ_HANDLER_FUNCTION(index)				\
 	.bitrate = DT_INST_PROP(index, clock_frequency),		\
 	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),			\
-	I2C_STM32_SCL_INIT(index)					\
-	I2C_STM32_SDA_INIT(index)					\
-	USE_TIMINGS(index)						\
+	IF_ENABLED(CONFIG_I2C_STM32_BUS_RECOVERY,			\
+		(.scl =	GPIO_DT_SPEC_INST_GET_OR(index, scl_gpios, {0}),\
+		 .sda = GPIO_DT_SPEC_INST_GET_OR(index, sda_gpios, {0}),))\
+	IF_ENABLED(DT_HAS_COMPAT_STATUS_OKAY(st_stm32_i2c_v2),		\
+		(.timings = (const struct i2c_config_timing *) i2c_timings_##index,\
+		 .n_timings = ARRAY_SIZE(i2c_timings_##index),))	\
```

Comment:
I would normally write this as

```c
IF_ENABLED(CONFIG_I2C_STM32_BUS_RECOVERY, (                     \
.scl = GPIO_DT_SPEC_INST_GET_OR(index, scl_gpios, {0}),         \
.sda = GPIO_DT_SPEC_INST_GET_OR(index, sda_gpios, {0}),         \
))                                                              \
IF_ENABLED(DT_HAS_COMPAT_STATUS_OKAY(st_stm32_i2c_v2), (        \
.timings = (const struct i2c_config_timing *)i2c_timings_##index, \
.n_timings = ARRAY_SIZE(i2c_timings_##index),                   \
))                                                              \
```

there's no hard guideline but that what looks more readable to me since the IF_ENABLED behaves like a `#ifdef` and we don't indent on that, and that would leave the field indented as they should rather than making it look they are nested in a lower level. How would you feel about that? @erwango 

blocking since you have a stray tab after `scl =`

Reply:
Actually, this was the initial indentation, which I requested to update to the current version.

Rationale was about readability, which isn't great in those sections since they are generally not covered by syntactic highlighting. To me adding indentation allows to better distinguish between conditions and actual structure assignments.

Reply:
cool by me

Reply:
...though it'd be great if you could fix that stray tab @ggrs, sorry I can't unsee it now :-)